win32/src/gballoc_hl_passthrough.c

```diff
@@ -0,0 +1,166 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include <stddef.h>
+
+#include "azure_macro_utils/macro_utils.h"
+#include "azure_c_logging/xlogging.h"
+
+#include "azure_c_pal/gballoc_ll.h"
+
+#include "azure_c_pal/gballoc_hl.h"
+
+static int wasInitialized = 0;
+
+int gballoc_hl_init(void* gballoc_hl_init_params, void* gballoc_ll_init_params)
+{
+    int result;
+    (void)gballoc_hl_init_params; /*are ignored, this is "passthrough*/
+
+    /*Codes_SRS_GBALLOC_HL_PASSTHROUGH_02_001: [ gballoc_hl_init shall call gballoc_ll_init(gballoc_ll_init_params). ]*/
+
+    if (gballoc_ll_init(gballoc_ll_init_params) != 0)
+    {
+        /*Codes_SRS_GBALLOC_HL_PASSTHROUGH_02_003: [ If there are any failures then gballoc_hl_init shall fail and return a non-zero value. ]*/
+        LogError("failure in gballoc_ll_init(gballoc_ll_init_params=%p)", gballoc_ll_init_params);
+        result = MU_FAILURE;
+    }
+    else
+    {
+        /*Codes_SRS_GBALLOC_HL_PASSTHROUGH_02_002: [ gballoc_hl_init shall succeed and return 0. ]*/
+        result = 0;
+        wasInitialized = 1;
+    }
+
+    return result;
+}
+
+void gballoc_hl_deinit(void)
+{
+    /*no work for this layer, it is passthough*/

```

Comment:
>/*no work for this layer, it is passthough*/ [](start = 4, length = 44)

nit: Comment sort of applies to everything here (technically this pass-through layer does add logging and even this function enforces calling init/deinit once). Maybe the comment is not really needed

Reply:
removed

---
In reply to: [459661312](https://github.com/Azure/azure-c-pal/pull/14#discussion_r459661312) [](ancestors = 459661312)